infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationDiscoveryService.java

```diff
@@ -328,17 +339,45 @@ private void connectToClusterManager() {
      * - Building Log Replication Context (LR context shared across receiving and sending components)
      * - Start Log Replication Server (receiver component)
      */
-    private void bootstrapLogReplicationService() {
+    private void bootstrapLogReplicationService(ClusterRole role) {
         log.info("Bootstrap the Log Replication Service");
+
         // Through LogReplicationConfigAdapter retrieve system-specific configurations
         // such as streams to replicate and version
-        LogReplicationConfig logReplicationConfig = getLogReplicationConfiguration(getCorfuRuntime());
+        LogReplicationConfig logReplicationConfig =
+            getLogReplicationConfiguration(getCorfuRuntime());
+
+        Set<String> remoteClusterIds = new HashSet<>();
 
-        logReplicationMetadataManager = new LogReplicationMetadataManager(getCorfuRuntime(),
-            topologyDescriptor.getTopologyConfigId(), localClusterDescriptor.getClusterId());
+        if (role == ClusterRole.ACTIVE) {
+            remoteClusterIds.addAll(topologyDescriptor.getStandbyClusters()
+                .keySet());
+        } else if (role == ClusterRole.STANDBY) {
+            remoteClusterIds.addAll(topologyDescriptor.getActiveClusters()
+                .keySet());
+        } else {
+            log.debug("Cluster role is {}.  Not bootstrapping further.", role);
+            return;
+        }
+
+        for (String remoteClusterId : remoteClusterIds) {
```

Comment:
This is somehow weird. If you think of it, in the case a node is the active, you're adding in advance one LogReplicationSinkManager per remote standby (this was done previously because there was only 1:1 instances of active:standby) so whenever  switchover was performed this took the other role and had the instance already created. However, in the scenario of multiple standby's, if there is a switchover, this guy will become the standby for only one active. So:

1. Only one instance of LogReplicationSinkManager is needed at a time, probably its better we dynamically created on role change (only when it's active)---before we did it in advance because of the 1:1 it was going to be the only option. 
2. In methods like: updateTopologyConfigId you are updating values for all sink managers, what does this mean in those cases?
3. In the case the role is standby, there can only be one active, so why do we expect to keep a collection of LRSM? If you can only be the Sink to one?

Reply:
hmm... I do see the issue here.. if the topology consists of 1 active/n standby's,  standby clusters will have only 1 remote(active cluster) and hence only 1 sink manager(SM).

After switchover, it will continue to have 1 SM when ideally there should have been more.

I think we should refresh/recreate the list of sink managers when the role changes.  I'll think more on these lines.